### DISCUSSION TITLE
Execute api call on drag event

### DISCUSSION THREAD

**Discussion by rudy1976s** [GrapesJS/grapesjs#5058] [Q&A]
Hello my name is Rudy, and I need to understand how to achieve the following result:
Create a block containing a list of posts ( there is API to get the list of the posts )
on dragging post from the above mentioned list, I need to be able to call a GET API which will get the markupof the post ( content ) and paste it to the canvas
Is it possible ?
Actually I am using a weird technique: I build the list with every content of the list already in the editor, getting data for every post when editor starts, buit is very slow indeed, I need to load only post at runtime when dragging .

Any help would be very appreciated!

Thanks in advance

**Comment by artf** [comment]
Hi Rudy, I created an example of how you could fetch dynamic content by using a custom component
https://jsfiddle.net/ha42ubq7/17/
The fetch is placed inside `view.onActive` and is triggered by using `activate: true` option on the block.
